src/ui/screens/profile/EditProfile.tsx

```diff
@@ -0,0 +1,148 @@
+import { RegularText } from '../../misc/text';
+import { getImageSource } from '../../../helpers/imageDataHelpers';
+import { TouchableView } from '../../../components/TouchableView';
+
+const openImagePicker = async (onUpdatePicture: (imageData: ImageData) => void) => {
+    const imageData = await AsyncImagePicker.showImagePicker();
+    if (imageData != null) {
+        onUpdatePicture(imageData);
+    }
+};
+
+const profileImageWidth = Dimensions.get('window').width * 0.6;
+
+export interface StateProps {
+    navigation: TypedNavigation;
+    profile: PublicProfile;
+    gatewayAddress: string;
+}
+
+export interface DispatchProps {
+    onUpdateAuthor: (text: string) => void;
+    onUpdatePicture: (image: ImageData) => void;
+}
+
+export const EditProfile = (props: StateProps & DispatchProps) => (
+    <FragmentSafeAreaViewWithoutTabBar>
+        <NavigationHeader
+            title='Edit profile'
+            navigation={props.navigation}
+        />
+        <ScrollView
+            keyboardShouldPersistTaps='handled'
+        >
+            <View style={styles.imagePickerContainer}>
+                <TouchableView
+                    onPress={async () => {
+                        await openImagePicker(props.onUpdatePicture);
+                    }}
+                >
+                    <ImageBackground
+                        source={getImageSource(
+                            props.profile.image,
+                            new ReactNativeModelHelper(props.gatewayAddress),
+                            defaultImages.defaultUser)
+                        }
+                        style={styles.imageBackground}
+                        imageStyle={styles.image}
+                    >
+                        <View style={styles.imagePickerIcon}>
+                            <Icon
+                                name={'pencil'}
+                                size={18}
+                                color={Colors.BLACK}
+                            />
+                        </View>
+                    </ImageBackground>
+                </TouchableView>
+            </View>
+            <RegularText style={styles.tooltip}>NAME</RegularText>
+            <SimpleTextInput
+                style={styles.row}
+                defaultValue={props.profile.name}
+                placeholder={DEFAULT_AUTHOR_NAME}
+                autoCapitalize='none'
+                autoFocus={props.profile.name === ''}
+                autoCorrect={false}
+                selectTextOnFocus={true}
+                returnKeyType={'done'}
+                onSubmitEditing={(name) =>
+                    name === ''
+                    ? props.onUpdateAuthor(DEFAULT_AUTHOR_NAME)
+                    : props.onUpdateAuthor(name)
+                }
+            />
+        </ScrollView>
+    </FragmentSafeAreaViewWithoutTabBar>
+);
+
+const WIDTH = Dimensions.get('window').width;
```

Comment:
it's advised to always get this value in the `render` function

Reply:
I read about this too, the main reason is when you support both portrait and landscape and you want to keep your app adaptive.

Here we use it for calculating style properties so those calculations should go to the render function. As we don't support landscape I didn't bother, also most of this component is a copy of the ProfileScreen from the onboarding.